Source/WTF/wtf/text/StringImpl.h

```diff
@@ -930,18 +931,22 @@ inline StringImpl::StringImpl(unsigned length)
     STRING_STATS_ADD_16BIT_STRING(m_length);
 }
 
-template<typename Malloc>
-inline StringImpl::StringImpl(MallocSpan<LChar, Malloc> characters)
-    : StringImplShape(s_refCountIncrement, { static_cast<const LChar*>(nullptr), characters.span().size() }, s_hashFlag8BitBuffer | StringNormal | BufferOwned)
+template<typename CharacterType, typename Malloc>
+std::span<const CharacterType> StringImpl::ensureStringImplMallocSpan(MallocSpan<CharacterType, Malloc>&& characters)
 {
     if constexpr (std::is_same_v<Malloc, StringImplMalloc>)
-        m_data8 = characters.leakSpan().data();
+        return characters.leakSpan();
```

Comment:
We can move-construct a MallocSpan return value here. That's a little clearer for a conversion function. (Then our StringImpl caller can leak the result, since StringImpl promises to delete in its destructor.)

Reply:
Makes sense. thank you for the thoughtful review.